src/kernel/capinv.c

```diff
@@ -1395,7 +1448,7 @@ composite_syscall_slowpath(struct pt_regs *regs, int *thd_switch)
 			vaddr_t frame_addr = __userregs_get1(regs);
 			paddr_t frame;
 
-			ret = pgtbl_get_cosframe(((struct cap_pgtbl *)ch)->pgtbl, frame_addr, &frame);
+			ret = pgtbl_get_cosframe(&(((struct cap_pgtbl *)ch)->pgtbl), frame_addr, &frame);
```

Comment:
I think the second option is more obvious, but it have some performance down sides.